model/src/test/scala/uk/gov/ons/addressIndex/model/AddressIndexModelTest.scala

```diff
@@ -7,34 +7,27 @@ class AddressIndexModelTest extends FlatSpec with Matchers {
 
   it should "produce the correct `AddressScheme` (`PostcodeAddressFile`) for the given string `paf`" in {
     val sample = "paf"
-    sample.toAddressScheme map (_ shouldBe a [PostcodeAddressFile])
+    sample.stringToScheme  shouldBe a [PostcodeAddressFile]
   }
 
   it should "produce the correct `AddressScheme` (`PostcodeAddressFile`) for the given string `postcodeAddressFile`" in {
     val sample = "postcodeAddressFile"
-    sample.toAddressScheme map (_ shouldBe a [PostcodeAddressFile])
+    sample.stringToScheme shouldBe a [PostcodeAddressFile]
   }
 
   it should "produce the correct `AddressScheme` (`BritishStandard7666`) for the given string `britishstandard7666`" in {
     val sample = "britishstandard7666"
-    sample.toAddressScheme map(_ shouldBe a [BritishStandard7666])
+    sample.stringToScheme shouldBe a [BritishStandard7666]
   }
 
   it should "produce the correct `AddressScheme` (`BritishStandard7666`) for the given string `bs7666`" in {
     val sample = "bs7666"
-    sample.toAddressScheme map(_ shouldBe a [BritishStandard7666])
+    sample.stringToScheme shouldBe a [BritishStandard7666]
   }
 
   it should "produce the correct `AddressScheme` (`BritishStandard7666`) for the given string `bs`" in {
     val sample = "bs"
-    sample.toAddressScheme map(_ shouldBe a [BritishStandard7666])
+    sample.stringToScheme shouldBe a [BritishStandard7666]
   }
 
```

Comment:
Where are the removed tests? I just removed the exception as it made the code more complex without a need.
You don't throw an exception when you get no element in an option, you return `None`. Here we return `UnsupportedSchema`